distribution/sdk/sample-cometd/cometd-java-example/src/main/java/org/codice/ddf/sdk/cometd/AsyncClient.java

```diff
@@ -0,0 +1,231 @@
+            public void onMessage(ClientSessionChannel channel, Message message) {
+                if (message.isSuccessful()) {
+                    asyncClientId = channel.getSession().getId();
+
+                    client.getChannel(ALL_ACTIVITIES).addListener(new ClientSessionChannel.MessageListener() {
+                        public void onMessage(ClientSessionChannel channel, Message message) {
+                            activities = message.getDataAsMap();
+                        }
+                    });
+
+                    client.getChannel(ALL_NOTIFICATIONS).addListener(new ClientSessionChannel.MessageListener() {
+                        public void onMessage(ClientSessionChannel channel, Message message) {
+                            notifications = message.getDataAsMap();
+                        }
+                    });
+                }
+            }
+        });
+        boolean handshaken = client.waitFor(1000, BayeuxClient.State.CONNECTED);
+
+        checkAllActivities();
+        checkAllNotifications();
+    }
+
+    /**
+     * Get all notification messages
+     * @return - notifications map
+     */
+    public Map<String, Object> getNotifications() {
+        return notifications;
+    }
+
+    /**
+     * Get all activities messages
+     * @return - activities map
+     */
+    public Map<String, Object> getActivities() {
+        return activities;
+    }
+
+    /**
+     * Get all query response messages
+     * @return - query response map
+     */
+    public Map<String, Object> getQueryResponse() {
+        return queryResponse;
+    }
+
+    /**
+     * Retrieves all persisted notifications
+     */
+    public void checkAllNotifications() {
+
+        client.getChannel(NOTIFICATIONS_CHANNEL).publish(emptyMessage);
+    }
+
+    /**
+     * Retrieves download notifications
+     */
+    public void checkDownloadNotifications() {
+
+        client.getChannel(DOWNLOADS_CHANNEL).publish(emptyMessage);
+    }
+
+    /**
+     * Retrieves all persisted activities
+     */
+    public void checkAllActivities() {
+
+        client.getChannel(ACTIVITIES_CHANNEL).publish(emptyMessage);
+    }
+
+    /**
+     * Execute a query against the ddf via the cometd endpoint
+     * @param keyword - Simple keyword for query
+     * @return - Query response map
+     * @throws InterruptedException
+     */
+    public void query(String keyword) throws InterruptedException {
+        String id = UUID.randomUUID().toString();
+        String responseChannel = "/" + id;
+        Map<String, Object> request = new HashMap<>();
+        request.put("id", id);
+        request.put("cql", "anyText ILIKE  '" + keyword + "'");
+
+        client.getChannel(responseChannel).addListener(new ClientSessionChannel.MessageListener() {
+            public void onMessage(ClientSessionChannel channel, Message message) {
+                queryResponse = message.getDataAsMap();
```

Comment:
We need to unregister after we receive a response otherwise we're leaking listeners. Either that or we need a single listener that can correlate between queries and responses.